spot_wrapper/wrapper.py

```diff
@@ -259,25 +442,90 @@ def _start_query(self):
             return callback_future
 
 
+def try_claim(func=None, *, power_on=False):
+    """
+    Decorator which tries to acquire the lease before executing the wrapped function
+
+    the _func=None and * args are required to allow this decorator to be used with or without arguments
+
+    Args:
+        func: Function that is being wrapped
+        power_on: If true, power on after claiming the lease
+
+    Returns:
+        Decorator which will wrap the decorated function
+    """
+    # If this decorator is being used without the power_on arg, return it as if it was called with that arg specified
+    if func is None:
+        return functools.partial(try_claim, power_on=power_on)
+
+    @functools.wraps(func)
+    def wrapper_try_claim(self, *args, **kwargs):
+        if self._get_lease_on_action:
```

Comment:
Note to Jenny: we will want to set `get_lease_on_action` true

Reply:
In my PR for the ros2 repo I made sure to initialise the wrapper with that parameter set to true: https://github.com/heuristicus/spot_ros2/blob/866098eccebaac9ac509f918b08293af020fb2cb/spot_driver/spot_driver/spot_ros2.py#L1020-L1021.